apps/condo/domains/contact/components/ContactsEditor.tsx

```diff
@@ -0,0 +1,566 @@
+    // Refs are used to keep it
+
+    const contactFieldsRef = useRef(contactFields)
+    useEffect(() => {
+        contactFieldsRef.current = contactFields
+    }, [contactFields])
+
+    const shouldCreateContactRef = useRef(shouldCreateContact)
+    useEffect(() => {
+        shouldCreateContactRef.current = shouldCreateContact
+    }, [shouldCreateContact])
+
+    const createContactAction = Contact.useCreate({ }, () => Promise.resolve())
+
+    const handleChangeContact = (values, isNew) => {
+        setContactFields(values)
+        setShouldCreateContact(isNew)
+    }
+
+    const createContact = async (organization, property, unitName) => {
+        if (shouldCreateContactRef.current) {
+            try {
+                return await createContactAction({
+                    ...contactFieldsRef.current,
+                    organization,
+                    property,
+                    unitName,
+                })
+            } catch (e) {
+                // Duplicated contacts should be figured out on the client,
+                // and "create" action should not be performed.
+                // In case of violation of unique constraint on `Contact` table,
+                // be silent for a user, but make a record in log.
+                if (e.message.match('Contact_uniq')) {
+                    console.error(e)
+                } else {
+                    throw (e)
+                }
+            }
+        }
+    }
+
+    const ContactsEditorComponent: React.FC<IContactEditorProps> = useMemo(() => {
+        const ContactsEditorWrapper = (props) => (
+            <ContactsEditor
+                {...props}
+                organization={organization}
+                onChange={handleChangeContact}
+            />
+        )
+        return ContactsEditorWrapper
+    }, [])
+
+    return {
+        createContact,
+        ContactsEditorComponent,
+    }
+}
+
+export const ContactsEditor: React.FC<IContactEditorProps> = (props) => {
+    const { form, fields, value: initialValue, onChange, organization, property, unitName } = props
+
+    const [contacts, setContacts] = useState([])
+    const [loading, setLoading] = useState(true)
+    const [error, setError] = useState()
+    const { link: { role } } = useOrganization()
+    const client = useApolloClient()
+
+    searchContacts(client, {
+        organizationId: organization,
+        propertyId: property ? property : undefined,
+        unitName : unitName ? unitName : undefined,
+    })
+        .then(({ data, loading, error }) => {
+            setContacts(data.objs)
+            setLoading(loading)
+            setError(error)
+        })
+
+    Contact.useObjects({
+        where: {
+            organization: { id: organization },
+            property: property ? { id: property } : undefined,
+            unitName: unitName || undefined,
+        },
+    }, {
+        fetchPolicy: 'network-only',
+    })
```

Comment:
what about return result?

Reply:
Oops, this is an orphaned code.
Removed in `293c5cc0`